### PR TITLE
Fix svg favicons

### LINKED ISSUE #2286: Favicons only offered in SVG format show as broken images in site dropdown
### Past Issues Searched

- [X] I have searched open and closed issues to make sure that the bug has not yet been reported

### Issue is a Bug Report

- [X] This is a bug report and not a feature request, nor asking for self-hosted support

### Using official Plausible Cloud hosting or self-hosting?

Plausible Cloud from plausible.io

### Describe the bug

Sites which only offer a favicon in SVG format, which is now accepted by all browsers, will appear as a broken image in the site selector dropdown.

### Expected behavior

Site icon is displayed.

### Screenshots

image

### Environment

_No response_

### PR DESCRIPTION
**Pull request by ukutaht**
### Changes

Fixes #2286

DDG favicon service does not handle SVG images well. Here's an SVG icon hosted at https://ptable.com/icon/pt-square-192.png which returns `content-type=image/svg+xml`. Requesting this through DDG:

```bash
➜  assets git:(fix-svg-favicons) curl -I https://icons.duckduckgo.com/ip3/ptable.com.ico
HTTP/2 200
server: nginx
date: Mon, 03 Oct 2022 08:33:30 GMT
content-type: image/x-icon
```

It has set `content-type=x-icon`. This incorrect content-type causes browsers to render a broken icon.

I also took the opportunity for some refactoring+cleanup in `assets/js/dashboard/site-switcher.js`. The diff is big because I ran the `prettier` formatter on the file, but this was done in a separate commit so you can still see the meaningful changes in [a commit diff.](https://github.com/plausible/analytics/pull/2295/commits/86ac9226bdaa966de3f17b161d3a88ee05d317af)

### Tests
- [x] Automated tests have been added

### Changelog
- [x] This PR does not make a user-facing change

### Documentation
- [x] This change does not need a documentation update

### Dark mode
- [x] This PR does not change the UI

### CODE REVIEW

**Review by vinibrsl — APPROVED** [review]

**Comment by aerosol** [comment]
> Here's an SVG icon hosted at https://ptable.com/icon/pt-square-192.png which returns content-type=image/svg+xml

err? 

```
$ curl -sI https://ptable.com/icon/pt-square-192.png | grep content-type
content-type: image/png
```

**Comment by aerosol** [comment]
I will let ddg know about this - it looks like a bug

image

File: lib/plausible_web/plugs/favicon.ex [hunk 1]
```diff
@@ -53,17 +53,26 @@ defmodule PlausibleWeb.Favicon do
   I'm not sure why DDG sometimes returns a broken PNG image in their response but we filter that out.
   When the icon request fails, we show a placeholder favicon instead. The placeholder is an emoji
   from https://favicon.io/emoji-favicons/
+
+  DuckDuckGo favicon service has some issues with [svg favicons](https://css-tricks.com/svg-favicons-and-all-the-fun-things-we-can-do-with-them/).
+  For some reason, they return then with `content-type=image/x-icon` whereas SVG icons should be returned with `conten-type=image/svg+xml`.
```
Comment (aerosol) [inline lib/plausible_web/plugs/favicon.ex:58]:
```suggestion
  For some reason, they return them with `content-type=image/x-icon` whereas SVG icons should be returned with `content-type=image/svg+xml`.
```

**Review by aerosol — APPROVED** [review]
LGTM, only typos spotted.

**Comment by ukutaht** [comment]
> > Here's an SVG icon hosted at https://ptable.com/icon/pt-square-192.png which returns content-type=image/svg+xml
> 
> err?
> 
> ```
> $ curl -sI https://ptable.com/icon/pt-square-192.png | grep content-type
> content-type: image/png
> ```

Eh... what? My bad. 

```bash
➜  analytics git:(master) curl  https://icons.duckduckgo.com/ip3/ptable.com.ico
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 121.1 121.1">
 <defs>
  <style>
   .Square	{ fill: url(#lgrad); }
   .Pt		{ fill: white; /* paint-order: stroke; stroke: rgb(255, 255, 255, 0.4); stroke-width: 2; */ }
  </style>
  <linearGradient id="lgrad" y2="100%">
   <stop offset="0" stop-color="rgb(45,190,252)" />
   <stop offset="1" stop-color="rgb(2,118,253)" />
  </linearGradient>
 </defs>
 <rect class="Square" width="121.1" height="121.1"/>
 <g class="Pt">
  <path d="M25,93.38V27.71H46.3q12.09,0,15.77,1a17.2,17.2,0,0,1,9.45,6.43q3.81,4.95,3.81,12.79a21.29,21.29,0,0,1-2.2,10.16,18,18,0,0,1-5.57,6.48,19.26,19.26,0,0,1-6.88,3.11,75.46,75.46,0,0,1-13.75.94H38.28V93.38ZM38.28,38.82V57.46h7.26c5.22,0,8.72-.35,10.48-1a8.69,8.69,0,0,0,4.14-3.23,8.8,8.8,0,0,0,1.5-5.1,8.52,8.52,0,0,0-2.1-5.92,9.18,9.18,0,0,0-5.33-2.91,63.38,63.38,0,0,0-9.54-.45Z"/>
  <path d="M104.27,45.81v10h-8.6V75a48.82,48.82,0,0,0,.24,6.78A2.75,2.75,0,0,0,97,83.39a3.58,3.58,0,0,0,2.13.63,16.14,16.14,0,0,0,5.06-1.21l1.08,9.77a25.11,25.11,0,0,1-10,1.88,15.76,15.76,0,0,1-6.13-1.14,9,9,0,0,1-4-3,11.73,11.73,0,0,1-1.77-4.91,61,61,0,0,1-.4-8.87V55.84H77.26v-10H83V36.36L95.67,29v16.8Z"/>
 </g>
</svg>
➜  analytics git:(master) curl -Is https://icons.duckduckgo.com/ip3/ptable.com.ico | grep content-type
content-type: image/x-icon
x-content-type-options: nosniff
```

Not sure how DDG gets SVG here but the content-type is clearly wrong  ¯\_(ツ)_/¯

**Comment by aerosol** [comment]
> Not sure how DDG gets SVG here but the content-type is clearly wrong ¯_(ツ)_/¯

I've let them know via their search feedback form

**Comment by ukutaht** [comment]
Thanks @aerosol !

**Pull request merged**